app/lib/frontend/backend.dart

```diff
@@ -269,6 +269,35 @@ class Backend {
       }
     }
   }
+
+  /// Updates flags on [package].
+  Future updateFlag(String package,
+      {bool isDiscontinued, bool doNotAdvertise}) async {
+    final pkgKey = db.emptyKey.append(models.Package, id: package);
+    await withAuthenticatedUser((user) async {
+      String latestVersion;
+      await db.withTransaction((tx) async {
+        final p = (await tx.lookup<models.Package>([pkgKey])).single;
+        if (p == null) {
+          throw Exception('Package does not exists.');
```

Comment:
Don't you need to throw an exception that indicates that we can show the message to the user..?

Otherwise, updating this flag for a package that doesn't exist will cause a 5xx, instead of a 4xx.

Reply:
Would this be a custom exception that we use for emitting 4xx with a custom message?